src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1002,34 +1002,62 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
        * have pkgs layered). This is still just a heuristic, since e.g. an InactiveRequest
        * may in fact become active in the new base, but we don't have the full tree. */
 
+      /* Note here that we use the cfg merge deployment for releasever: the download
+       * metadata only path is currently used only by the auto-update checker, and there
+       * we want to show updates/vulnerabilities relative to the *booted* releasever. */
+      g_autoptr(OstreeDeployment) cfg_merge_deployment =
+        ostree_sysroot_get_merge_deployment (sysroot, self->osname);
+
+      g_autoptr(DnfSack) sack = NULL;
+
       /* XXX: in jigdo mode we'll want to do this unconditionally */
       if (g_hash_table_size (rpmostree_origin_get_packages (origin)) > 0)
         {
           /* XXX: dedupe a bit more with RefreshMd path */
           g_autoptr(RpmOstreeContext) ctx =
             rpmostree_context_new_system (repo, cancellable, error);
 
-          /* Note here that we use the cfg merge deployment for releasever: the download
-           * metadata only path is currently used only by the auto-update checker, and there
-           * we want to show updates/vulnerabilities relative to the *booted* releasever.
-           * Anyway, given that we don't yet do etc merges on boot, it shouldn't be too
-           * common for users to stay long on e.g. f26 when they have f27 already deployed
-           * and ready to reboot into. */
-          g_autoptr(OstreeDeployment) cfg_merge_deployment =
-            ostree_sysroot_get_merge_deployment (sysroot, self->osname);
-
           g_autofree char *source_root =
             rpmostree_get_deployment_root (sysroot, cfg_merge_deployment);
           if (!rpmostree_context_setup (ctx, NULL, source_root, NULL, cancellable, error))
             return FALSE;
 
           /* we always want to force a refetch of the metadata */
-          dnf_context_set_cache_age (rpmostree_context_get_dnf (ctx), 0);
+          DnfContext *dnfctx = rpmostree_context_get_dnf (ctx);
+          dnf_context_set_cache_age (dnfctx, 0);
 
           /* point libdnf to our repos dir */
           rpmostree_context_configure_from_deployment (ctx, sysroot, cfg_merge_deployment);
 
-          if (!rpmostree_context_download_metadata (ctx, cancellable, error))
+          /* streamline: we don't need rpmdb or filelists, but we *do* need updateinfo */
+          if (!rpmostree_context_download_metadata (ctx,
+                DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_RPMDB |
+                DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS |
```

Comment:
In non-rojig mode we need the filelists to depsolve in general, not downloading them when someone does `refresh-md` to me seems...confusing at best.

Reply:
This isn't the `refresh-md` path, though. We do download everything there.

This path is strictly for `check` mode. I'm open to downloading filelists here too, though given that checks will run much more frequently than any operations that needs filelists, I opted for making the `check` as lightweight as possible. WDYT?